webauthn-server-core/src/main/java/com/yubico/webauthn/RelyingParty.java

```diff
@@ -385,6 +394,15 @@ private RelyingParty(
     ) {
         this.identity = identity;
         this.origins = origins != null ? origins : Collections.singleton("https://" + identity.getId());
+
+        for (String origin : this.origins) {
+            try {
+                new URL(origin);
+            } catch (MalformedURLException e) {
+                log.warn("Allowed origin is an invalid URL, it will match only by exact string equality: {}", origin);
+            }
+        }
```

Comment:
With the check in the constructor, the warning will (probably) be printed immediately on server startup, instead of being deferred to when a user performs a registration or authentication ceremony. This does duplicate the warning in `OriginMatcher`, but by the "fail faster" ethos I think it's worthwhile to warn as early as possible.

Reply:
Ok